addons/packages/harbor/2.3.3/test/e2e/harbor_suite_test.go

```diff
@@ -173,11 +174,34 @@ func findPackageAvailableVersion(packageName string, versionSubstr string) strin
 	return matchedVersions[len(matchedVersions)-1]
 }
 
-func hasDefaultStorageClass() bool {
-	output, err := utils.Kubectl(nil, "get", "storageclasses")
+/*
+apply a "rancher.io/local-path" storageclass for clusters lack of csi driver to dynamically provisioning for PVC usages
+clusters included and before tkg1.5(k8s1.22) have in-tree cloud provider plugin
+vsphere clusters have own storageclass with csi "csi.vsphere.vmware.com"
+
+only when clusters without csi driver but with annotations "migrated_plugins" will have to install local-path-storage
+aws and azure clusters would apply this "rancher.io/local-path" sc instead of the default one
+
+return storageclass name for PVC
+*/
+func setStorageClass() string {
+	jsonPath := `jsonpath='{.items[0].metadata.annotations}'`
+	csiMigratedPlugins, err := utils.Kubectl(nil, "get", "csinodes", "-o", jsonPath)
+	fmt.Println("csiMigratedPlugins:", csiMigratedPlugins)
+	Expect(err).NotTo(HaveOccurred())
+
+	jsonPath = `jsonpath='{.items[0].spec.drivers}'`
+	csidriver, err := utils.Kubectl(nil, "get", "csinodes", "-o", jsonPath)
+	fmt.Println("csidriver:", csidriver)
 	Expect(err).NotTo(HaveOccurred())
 
-	return strings.Contains(output, "(default)")
+	if (csidriver == "'<nil>'") && (strings.Contains(csiMigratedPlugins, "kubernetes.io/aws-ebs,kubernetes.io/azure-disk")) {
+		// apply a new local-path-storage
+		_, err := utils.Kubectl(nil, "apply", "-f", filepath.Join("fixtures", "local-path-storage.yaml"))
+		Expect(err).NotTo(HaveOccurred())
+		return "local-path"
+	}
+	return ""
```

Comment:
Please change the logic to get the default storage class first, then verify it's available. If there isn't a default storage class or it isn't available, install the local path and return the local path as the storage when the local path is running.